conans/test/functional/toolchains/cmake/test_cmake_toolchain.py

```diff
@@ -1386,14 +1392,36 @@ def package(self):
                 cmake = CMake(self)
                 build_type = "Release" if cmake.is_multi_configuration else None
                 cmake.install(build_type=build_type)
-        """)
+        """.format(conanfile_settings, tc_cache_variables))
 
     cmake = """
         cmake_minimum_required(VERSION 3.15)
         project(pkg LANGUAGES NONE)
+        message(STATUS "CMAKE_BUILD_TYPE: '${CMAKE_BUILD_TYPE}'")
```

Comment:
This doesn't seem a valid check. CMAKE_BUILD_TYPE must be not defined for all multi-config generators, irrespective of all the inputs, toolchain, command line, etc.